### ISSUE TITLE
google_notebooks_instance don't persist tags in state

### ISSUE THREAD

**Issue by t3hami** [hashicorp/terraform-provider-google#17075]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```
~$ terraform -v
Terraform v1.6.4
on darwin_arm64
```

### Affected Resource(s)

* google_notebooks_instance

### Terraform Configuration Files

main.tf
```
resource "google_notebooks_instance" "instance" {
  name         = var.name
  project      = var.project
  location     = var.location
  machine_type = var.machine_type
  vm_image {
    project      = local.vm_image_project
    image_family = local.image_family
  }
  shielded_instance_config {
	  enable_secure_boot = var.enable_secure_boot
  }
  service_account = var.service_account

  install_gpu_driver = var.install_gpu_driver
  boot_disk_type     = var.boot_disk_type
  boot_disk_size_gb  = var.boot_disk_size_gb

  no_public_ip    = true
  no_proxy_access = false

  kms_key = var.kms_enabled == true ? data.google_kms_crypto_key.crypto_key[0].id : ""

  network = var.network
  subnet  = var.subnet
  tags    = var.tags

  labels = var.labels

  metadata = {
    proxy-mode = "service_account"
    terraform  = "true"
  }
}
```

```tf
# Copy-paste your Terraform configurations here.
#
# For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file.
# For security, you can also encrypt the files using our GPG public key:
#    https://www.hashicorp.com/security
#
# If reproducing the bug involves modifying the config file (e.g., apply a config,
# change a value, apply the config again, see the bug), then please include both:
# * the version of the config before the change, and
# * the version of the config after the change.
```

### Expected Behavior

Show no change when we run terraform apply again if we don't change anything

### Actual Behavior

Do not recreate vertex ai notebook. But due to network tags not persisting in state it is always recreating.

### Steps to Reproduce

Create any vertex ai notebook with network tags, it's not going to persist tags details.

### References

Similar issue which was closed without any fix https://github.com/hashicorp/terraform-provider-google/issues/9273 

* #0000

b/323154198

**Comment by edwardmedia** [comment]
@t3hami can you share your config along with the debug log? What version of the provider are you using?

**Comment by edwardmedia** [comment]
@t3hami is this still an issue?

**Comment by t3hami** [comment]
@edwardmedia I've updated the description with terraform main.tf. I'm using google provider 5.13.0

**Comment by edwardmedia** [comment]
@t3hami can you hardcode the config so I can apply?

**Comment by t3hami** [comment]
Yes @edwardmedia, exactly. Do we have any work around for this, until the issue is fixed?

**Comment by t3hami** [comment]
I tried terraform state refresh and also manually added tags in the state directly. But no luck with both tricks. It's always recreating and not able to see tags.

**Comment by t3hami** [comment]
This issue is still not fixed in the 5.17.0

**Comment by bcreddy-gcp** [comment]
It's a fix on the backend which is blocked by the v5.18.0 release.

**Comment by t3hami** [comment]
This issue is still not fixed in the 5.18.0

**Comment by bcreddy-gcp** [comment]
Could you check now @t3hami.
